neuro_san/http_sidecar/handlers/base_request_handler.py

```diff
@@ -124,6 +131,28 @@ def get_concierge_grpc_session(self, metadata: Dict[str, Any]) -> ConciergeSessi
                 metadata=metadata)
         return grpc_session
 
+    async def update_agents(self, metadata: Dict[str, Any]) -> bool:
+        """
+        Update internal agents table by executing request
+        to underlying gRPC service.
+        :param metadata: metadata for request from caller context.
+        :return: True if update was successful
+                 False otherwise
+        """
+        try:
+            data: Dict[str, Any] = {}
+            grpc_session: ConciergeSession = self.get_concierge_grpc_session(metadata)
+            agents_dict: Dict[str, Any] = grpc_session.list(data)
+            agents_list = agents_dict.get("agents", [])
+            agents_names: List[str] = []
+            for agent_dict in agents_list:
+                agents_names.append(agent_dict["agent_name"])
+            self.agents_updater.update_agents(agents_names)
+            return True
+        except Exception as exc:  # pylint: disable=broad-exception-caught
+            self.process_exception(exc)
+            return False
```

Comment:
In the near future we will have agents that are available to serve that are need-to-know and not listed as part of the concierge service.  There is a ticket for this already.  Might be worth a comment that this coupling will need to be revisited when that happens.

That all said it's a good thing that there is a single source of truth in the gRPC services for now.
I see why you were asking about "Can we get rid of gRPC?" now.

We might be able to devise a way so that this update doesn't have to happen as often.
It's possible similar mechanisms might need to be in play when we get to other protocol compliance like MCP.

Also FWIW: We probably do not need metadata for this internal/implementation-detail query.